### PR TITLE
Send pending alarms message even when empty

### PR DESCRIPTION
**Pull request by peterbryntesson**
fix: Added code to send pending alarms message even though the dictionary is empty

### CODE REVIEW

**Review by simonjaeger — APPROVED** [review]
LGTM

**Comment by peterbryntesson** [comment]
UPDATE: Need to fix one more issue that Trumpf found. Don't approve just yet.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/ISubscription.cs [hunk 1]
```diff
@@ -108,5 +108,10 @@ public interface ISubscription : IDisposable {
         /// </summary>
         /// <returns></returns>
         Task CloseAsync();
+
+        /// <summary>
+        ///
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/ISubscription.cs:113]: Please fill in.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs [hunk 2]
```diff
@@ -554,7 +554,7 @@ public class DefaultSessionManager : ISessionManager, IDisposable {
 
                 // Validate thumbprint
                 if (e.Certificate.RawData != null && !string.IsNullOrWhiteSpace(e.Certificate.Thumbprint)) {
-                    
+
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs:557]: Is this an empty line or is it my view?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs [hunk 3]
```diff
@@ -869,7 +869,24 @@ private sealed class SessionWrapper {
...
+                }
+                set {
+                    if (value != _state) {
+                        var oldState = _state;
+                        _state = value;
+
+                        if ((value == SessionState.Running && oldState != SessionState.Running) ||
+                            (value != SessionState.Running && oldState == SessionState.Running)) {
+                            foreach (var subscription in Subscriptions.Values) {
+                                subscription.OnSubscriptionStateChanged((bool)(value == SessionState.Running));
```
Comment (simonjaeger) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs:884]: This has to be cast to a bool?

**Review by simonjaeger — CHANGES_REQUESTED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/ISubscription.cs (hunk 1 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/ISubscription.cs:113]: Fixed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs (hunk 2 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs:557]: It was. Deleted

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs (hunk 3 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs:884]: Remnant of another try. Removed.

**Review by simonjaeger — APPROVED** [review]

**Review by mregen — APPROVED** [review]

**Pull request merged**